src/chain/store/chain_store.rs

```diff
@@ -625,8 +626,26 @@ where
 
         let mut writer = writer.lock().await;
         let digest = match &mut *writer {
-            Either::Left(left) => left.get_mut().finalize().await,
-            Either::Right(right) => right.finalize().await,
+            Either::Left(left) => {
+                left.flush()
```

Comment:
Can we define `AsyncWrite` for `Either`? Then we could just flush the `writer` instead of flushing `left` and `right`.

Reply:
Yeah, that would be nice and dry.
Unfortunately we'd still have to do the `finalize` separately in this case.

Reply:
I have introduced this change, turned out to be a little less involved than I thought it would.